opensrp-reveal/src/main/java/org/smartregister/reveal/fragment/AvailableOfflineMapsFragment.java

```diff
@@ -0,0 +1,251 @@
+        double topLeftLng = -119.5073;
+        double bottomRightLat = 37.6744;
+        double bottomRightLng = -119.6815;
+        double topRightLat = 37.7897;
+        double topRightLng = -119.6815;
+        double bottomLeftLat = 37.6744;
+        double bottomLeftLng = -119.5073;
+
+        String mapName = "Op Area 1";
+
+        currentMapDownload = mapName;
+
+        String mapboxStyle = "mapbox://styles/ona/cj9jueph7034i2rphe0gp3o6m";
+        LatLng topLeftBound = new LatLng(topLeftLat, topLeftLng);
+        LatLng topRightBound = new LatLng(topRightLat, topRightLng);
+        LatLng bottomRightBound = new LatLng(bottomRightLat, bottomRightLng);
+        LatLng bottomLeftBound = new LatLng(bottomLeftLat, bottomLeftLng);
+
+        double maxZoom = 20.0;
+        double minZoom = 0.0;
+
+        OfflineServiceHelper.ZoomRange zoomRange = new OfflineServiceHelper.ZoomRange(minZoom, maxZoom);
+
+        OfflineServiceHelper.requestOfflineMapDownload(getActivity()
+                , mapName
+                , mapboxStyle
+                , BuildConfig.MAPBOX_SDK_ACCESS_TOKEN
+                , topLeftBound
+                , topRightBound
+                , bottomRightBound
+                , bottomLeftBound
+                , zoomRange
+        );
+
+    }
+
+    private void stopMapDownLoad(@NonNull String mapName) {
+        OfflineServiceHelper.stopMapDownload(getActivity(), mapName , BuildConfig.MAPBOX_SDK_ACCESS_TOKEN);
+    }
+
+    private class MapDownloadReceiver extends BroadcastReceiver {
+        @Override
+        public void onReceive(Context context, Intent intent) {
+            Bundle bundle = intent.getExtras();
+            if (bundle != null) {
+                Log.i("KUJAKU SAMPLE APP TAG", intent.getExtras().toString());
+                if (bundle.containsKey(MapboxOfflineDownloaderService.KEY_RESULT_STATUS)
+                        && bundle.containsKey(MapboxOfflineDownloaderService.KEY_RESULT_MESSAGE)
+                        && bundle.containsKey(MapboxOfflineDownloaderService.KEY_RESULTS_PARENT_ACTION)
+                        && bundle.containsKey(Constants.PARCELABLE_KEY_MAP_UNIQUE_NAME)) {
+
+                    String mapUniqueName = bundle.getString(Constants.PARCELABLE_KEY_MAP_UNIQUE_NAME);
+                    String resultStatus = bundle.getString(MapboxOfflineDownloaderService.KEY_RESULT_STATUS);
+                    MapboxOfflineDownloaderService.SERVICE_ACTION serviceAction = (MapboxOfflineDownloaderService.SERVICE_ACTION) bundle.get(MapboxOfflineDownloaderService.KEY_RESULTS_PARENT_ACTION);
+
+                    String message = bundle.getString(MapboxOfflineDownloaderService.KEY_RESULT_MESSAGE);
+
+                    if (MapboxOfflineDownloaderService.SERVICE_ACTION_RESULT.FAILED.name().equals(resultStatus)) {
+                        if (!TextUtils.isEmpty(message)) {
+                            if (!message.contains("MapBox Tile Count limit exceeded")) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [These nested if statements could be combined](https://app.codacy.com/manual/OpenSRP/opensrp-client-reveal/pullRequest?prid=4217760)